### PR TITLE
Update build for numpy 2 with new methods

### PR DESCRIPTION
**Pull request by ChrisBarker-NOAA**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

@conda-forge-admin, please rerender

### CODE REVIEW

**Comment by ChrisBarker-NOAA** [comment]
@conda-forge/help-c-cpp

This seems to be building for numerous numpy versions -- rather than for 2.0 -- how to I get the "new" 2.0 builds?

**Comment by xhochy** [comment]
You need to add the numpy 2.0 migration file to `.ci_support/migrations` and then rerender.

**Comment by ChrisBarker-NOAA** [comment]
@conda-forge-admin, please rerender

**Comment by ChrisBarker-NOAA** [comment]
Finally! thanks @xhochy !

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,31 +10,45 @@ source:
...
     - python                                 # [build_platform != target_platform]
     - cross-python_{{ target_platform }}     # [build_platform != target_platform]
     - pybind11                               # [build_platform != target_platform]
+
     - {{ compiler('cxx') }}
+    - {{ stdlib('c') }} # this fails for me :-(
     - cmake
-    - python  # [build_platform != target_platform]
     - cross-python_{{ target_platform }}  # [build_platform != target_platform]
+    - numpy
```
Comment (jaimergp) [inline recipe/meta.yaml]:
```suggestion
    - numpy  # [build_platform != target_platform]
```

**Comment by ChrisBarker-NOAA** [comment]
Thanks! still confused about where you need:

`# [build_platform != target_platform]`

I would have thought that that's what putting it in host meant, but what do I know?

**Pull request merged**